modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/fetch.es.js

```diff
@@ -64,7 +64,7 @@ export default function defaultFetch(resource, init = {}) {
 				resource = resourceLocation;
 			}
 			else {
-				resource = {...resource, url: resourceLocation};
+				resource.url = resourceLocation;
```

Comment:
From just a code level view, this doesn't seem like it would be the right solution because these before/after are essentially equivalent. The previous change is just cloning all the properties of `resource`, so the same values should be there.

Reply:
Not really. I thought the same, but there are cases where it is not equivalent, I think. Namely, when `resource` is neither a String, nor an Object. Example:

![image](https://github.com/liferay-frontend/liferay-portal/assets/3273630/9863fbad-f34b-4a78-a423-56c3a2cb4841)

So I guess this failed for resources that are numbers or even objects that override the `toString()` method (if `fetch` uses it to compute what to fetch, which I'm not sure) and lose the override with the reassignment.

Additionally, the new code makes much more sense than the old one, because creating a new object just to set a property doesn't seem very efficient.

Unless, the new object was created on purpose to avoid modifying the original passed-in object. If that's the case, this fix could lead to different errors... :thinking:

Reply:
@victorcdp Do you know what was being fed to `fetch` when the error happened? Was it a pure object or something more exotic?

Reply:
It was an object, specifically a [URL typescript interface](https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.url.html), and it was being entirely overwritten and just the "url" remained, causing the fetch to fail. With this change, it just adds/updates the url property.

Reply:
@victorcdp if `resource`  is an object, the existing code `{...resource, url: resourceLocation};` would not overwrite, it would copy the existing keys into the new object. I suspect something else is going on

Reply:
I know, i found it weird too, but we couldn't figure out why it was happening, if you have any insights please let me know. [this](https://files.slack.com/files-pri/T03BTCQAJ-F05AUGJ25MW/image.png) is the object before, and [this](https://files.slack.com/files-pri/T03BTCQAJ-F05A4RKM8RG/image.png) is the object after.

Reply:
@bryceosterhaus , @izaera , what is the recommended step going forward for @victorcdp ?

Reply:
@victorcdp neither of those links work for me for some reason, it errors saying too many re-directs. Can you add the images here?


My concern here is that we don't know why this solution is fixing the bug. Because this the changes are essentially equivalent(assuming `resource` is an object). My worry is if we commit this change here without knowing why it is actually fixing the issue, then later on someone may refactor this code back to what we had previously and then get this bug again. 

So before I am comfortable with a proposed solution, I would want to know _why_ it is actually fixing the issue at hand, because I suspect that if we don't know why this is solving the bug then this might not actually be the root of the issue.

Reply:
This is before: ![before](https://github.com/liferay-frontend/liferay-portal/assets/63615316/1f18f723-379e-4c67-8ed1-a97a234a05e7)
and this is after:
![after](https://github.com/liferay-frontend/liferay-portal/assets/63615316/ae7ce3dd-675c-4156-894e-81c5afa85075)

Reply:
Oh interesting... The initial resource is of type `new URL()` which I would have thought would clone with an object destructure, but it looks like it doesn't. After looking into this I guess it is because "Properties on URL object are not enumerable." Meaning you wouldn't be able to do something like `Object.keys(myURL)` either.

I think the proper solution here then is to use `new URL(url)` because we know that `resource` is used on line 77 in the `fetch()` and fetch expects either a string or URL object. I would also make this change on line 41.

Let me know what you think and if that works as expected.

Reply:
> which I would have thought would clone with an object destructure, but it looks like it doesn't.

:thinking: Even if it cloned correctly, there are still possible pitfalls, I think...

Say you create a new class, point its prototype's toString()  to some custom method, then clone an instance of that type with the destructuring operator. The new instance won't have the correct toString() implementation (and this can happen with more properties/method probably).

In general, I think you need to specify (for any API) what type of objects you are expecting to receive and treat them accordingly (TypeScript would be quite useful here). We could even add preventive checks at the start of the method.

For example: if we only expect real `Objects`, we check for it and throw instead of failing silently with a hard-to-debug error...

Reply:
@izaera , if you have a solution, if the solution is going to require further discussion, I suggest the front-end team takes over the LPS ticket and work accordingly.